### ISSUE TITLE
Wrong procedure declaration generated in MySQL when using IN parameters with VARCHAR(20) types

### ISSUE THREAD

**Issue by AnghelLeonard** [jOOQ/jOOQ#12315]
### Expected behavior
No errors.

### Actual behavior
In MySQL, `createProcedure()` may lead to an error: "You have an error in your SQL syntax...".

### Steps to reproduce the problem
I tried this problem:

```java
Parameter<String> in_office_code = in("in_office_code", VARCHAR);

ctx.dropProcedureIfExists("get_emps_in_office_jooq")
      .execute();

ctx.createProcedure("get_emps_in_office_jooq")
      .parameters(in_office_code)
      .as(begin(
              select(OFFICE.CITY, OFFICE.COUNTRY, OFFICE.INTERNAL_BUDGET)
                                .from(OFFICE)
                                .where(OFFICE.OFFICE_CODE.eq(in_office_code)),
              select(EMPLOYEE.EMPLOYEE_NUMBER, EMPLOYEE.FIRST_NAME, EMPLOYEE.LAST_NAME)
                                .from(EMPLOYEE)
                                .where(EMPLOYEE.OFFICE_CODE.eq(in_office_code))
     ))
    .execute();
```

The rendered SQL contains syntax errors:

```sql
create procedure `get_emps_in_office_jooq`(`in_office_code` varchar) 
begin select `classicmodels`.`office`.`city`, `classicmodels`.`office`.`country`, 
`classicmodels`.`office`.`internal_budget` from `classicmodels`.`office` 
where `classicmodels`.`office`.`office_code` = `in_office_code`; 
select `classicmodels`.`employee`.`employee_number`, `classicmodels`.`employee`.`first_name`, 
`classicmodels`.`employee`.`last_name` from `classicmodels`.`employee` 
where `classicmodels`.`employee`.`office_code` = `in_office_code`; 
end;
```
In order to work I done this:

```sql
**DELIMITER $$;**
create procedure `get_emps_in_office_jooq`(`in_office_code` **varchar(20)**) 
begin select `classicmodels`.`office`.`city`, `classicmodels`.`office`.`country`, 
`classicmodels`.`office`.`internal_budget` from `classicmodels`.`office` 
where `classicmodels`.`office`.`office_code` = `in_office_code`; 
select `classicmodels`.`employee`.`employee_number`, `classicmodels`.`employee`.`first_name`, 
`classicmodels`.`employee`.`last_name` from `classicmodels`.`employee` 
where `classicmodels`.`employee`.`office_code` = `in_office_code`; 
end**$$**
**DELIMITER ;**
```

About rendering `VARCHAR(20)` I tried this:

```java
Parameter<String> in_office_code = in("in_office_code", VARCHAR(20));
```

But, it still renders only `VARCHAR` without length.

### Versions

- jOOQ:3.15.0
- Java:14
- Database (include vendor):MySQL
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your report. I think that's the same issue as https://github.com/jOOQ/jOOQ/issues/12297 ? It should work if you specify `allowMultiQueries=true` in your connection URL. Of course, we should get this to work without that flag, using your `DELIMITER` workaround.

I'll change this issue to be about the remaining `VARCHAR` vs `VARCHAR(20)` syntax only. It's usually good not to mix problems in issues, so they can be tracked individually. E.g. the `VARCHAR` vs `VARCHAR(20)` problem should be quite easy to fix, but the other is a bit more laborious, and won't be backported to jOOQ 3.15 (see explanation in https://github.com/jOOQ/jOOQ/issues/12297)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12321). Thanks again for your report!
